docs(kubernetes): version support policy MTA-1717 (#524)
scaleway/docs-content: compute/kubernetes/reference-content/version-support-policy.mdx

---
meta:
  title: Kubernetes version support policy
  description: This page explains the Kubernetes version support policy
content:
  h1: Kubernetes version support policy
  paragraph: This page explains the Kubernetes version support policy
dates:
  validation: 2022-04-04
  posted: 2022-04-04
categories: 
  - compute
  - kubernetes
---

[Kubernetes (K8s)](https://kubernetes.io) is an open source project that is constantly evolving, integrating and deprecating features and applying bug fixes. The project is supported by the community, which releases between four and six new minor versions every year.

> Scaleway Kubernetes team allows the usage of every new Kubernetes version as soon as the version is officially released and certified by the [Cloud Native Computing Foundation](https://www.cncf.io/), which usually makes the versions available in Kubernetes Kapsule and Kubernetes Kosmos within a few days after the release.

Reviewer: ```suggestion
With Scaleway Kubernetes you can use the latest version of Kubernetes as soon as the it is officially released and certified by the [Cloud Native Computing Foundation](https://www.cncf.io/). The versions are usually available in Kubernetes Kapsule and Kubernetes Kosmos within a few days after the release.
```

> Scaleway Kubernetes team allows the usage of every new Kubernetes version as soon as the version is officially released and certified by the [Cloud Native Computing Foundation](https://www.cncf.io/), which usually makes the versions available in Kubernetes Kapsule and Kubernetes Kosmos within a few days after the release.

Reviewer 2: on met un lien vers kosmos et kapsule?

Author: done

> Each minor Kubernetes version is supported by Scaleway Kubernetes products for at least twelve month before being deprecated.

Reviewer: ```suggestion
Each minor Kubernetes version is supported by Scaleway Kubernetes for at least twelve months before being deprecated.
```

> ## Version management recommandation

Reviewer 2: est-ce que ca on le mettrait pas dans un important ? au lieu qu'il ait sa propre section ?

> It is recommended that every Kubernetes cluster is maintained up-to-date with the latest minor version of Kubernetes available and the latest patch.

Reviewer: ```suggestion
We recommended you keep your Kubernetes clusters up-to-date with the latest available minor version of Kubernetes and the latest patches.
```

### Cluster auto-upgrade
The auto-upgrade functionality provided by Kubernetes Kapsule & Kubernetes Kosmos on the control plane and the managed nodes versions of Kubernetes can be activated using the API, CLI, or the Scaleway Console. For this functionality to operate correctly, the `kube-system` namespace of a cluster must not be altered or have any user-induced behaviors that could interfere with the auto-upgrade functionality.

## Support
Before opening a support ticket related to a cluster behavior or a Kubernetes component, we highly recommend that the cluster is upgraded to the latest version of Kubernetes. If the upgrade to a new minor version is not possible, the latest patch of the version running on the cluster must be applied.
80% of Kubernetes issues are usually fixed when applying the latest patch version available.

## Scaleway Kubernetes Kapsule & Kosmos release calendar

* **Release:** Version is available for cluster creation and cluster upgrade.

> * **Deprecation:** Version is removed from cluster creation option. Customers are contacted via a support ticket that they are using a deprecated version of Kubernetes and are asked to update their clusters.

Reviewer 2: ```suggestion
* **Deprecation:** Version is removed from cluster creation option. Customers are informed via a support ticket that they are using a deprecated version of Kubernetes and are asked to update their clusters.
```

|--------------------|------------------|----------------------|------------------------------|------------------|-------------------|
| 1.18               | March 23, 2020   | June 2021            | March 27, 2020               | February 9, 2022 | April 15, 2022    |
| 1.19               | August 26, 2020  | October 2021         | August 31, 2020              | August 2022      | October 15, 2022  |
| 1.20               | December 8, 2020 | February 2022        | December 8, 2020             | November 2022    | January 15, 2023  |
| 1.21               | April 8, 2021    | June 2022            | April 30, 2021               | April 2023       | June 15, 2023     |
| 1.22               | August 4, 2021   | October 2022         | August 11, 2021              | August 2023      | October 15, 2023  |
| 1.23               | December 8, 2021 | February 2023        | December 9, 2021             | December 2023    | February 15, 2024 |
| 1.24               | April 2022       |                      |                              |                  |                   |

## Version information

> For a complete changelog update, please check the official [Kubernetes changelog](https://kubernetes.io/releases/).

Reviewer: ```suggestion
For a complete changelog update, refer to the official [Kubernetes changelog](https://kubernetes.io/releases/).
```

---
meta:
  title: Kubernetes version support policy
  description: This page explains the Kubernetes version support policy
content:
  h1: Kubernetes version support policy
  paragraph: This page explains the Kubernetes version support policy
dates:
  validation: 2022-04-04
  posted: 2022-04-04
categories: 
  - compute
  - kubernetes
---

[Kubernetes (K8s)](https://kubernetes.io) is an open source project that is constantly evolving, integrating and deprecating features and applying bug fixes. The project is supported by the community, which releases between four and six new minor versions every year.

> With Scaleway Kubernetes you can use the latest version of Kubernetes as soon as the it is officially released and certified by the [Cloud Native Computing Foundation](https://www.cncf.io/). The versions are usually available in Kubernetes Kapsule and Kubernetes Kosmos within a few days after the release.

Reviewer 3: ```suggestion
With Scaleway Kubernetes you can use the latest version of Kubernetes as soon as it is officially released and certified by the [Cloud Native Computing Foundation](https://www.cncf.io/). The versions are usually available in Kubernetes Kapsule and Kubernetes Kosmos within a few days after the release.
```

<Message type="important">

Each minor Kubernetes version is supported by Scaleway Kubernetes for at least twelve months before being deprecated.

</Message>

## Version management recommandation
We recommended you keep your Kubernetes clusters up-to-date with the latest available minor version of Kubernetes and the latest patches.

### Cluster auto-upgrade

> The auto-upgrade functionality provided by [Kubernetes Kapsule(https://www.scaleway.com/en/kubernetes-kapsule/) & [Kubernetes Kosmos](https://www.scaleway.com/en/kubernetes-kosmos/) on the control plane and the managed node versions of Kubernetes can be activated using the API, CLI, or the Scaleway Console. For this functionality to operate correctly, the `kube-system` namespace of a cluster must not be altered or have any user-induced behaviors that could interfere with the auto-upgrade functionality.

Reviewer 3: ```suggestion
The auto-upgrade functionality provided by [Kubernetes Kapsule](https://www.scaleway.com/en/kubernetes-kapsule/) & [Kubernetes Kosmos](https://www.scaleway.com/en/kubernetes-kosmos/) on the control plane and the managed node versions of Kubernetes can be activated using the API, CLI, or the Scaleway console. For this functionality to operate correctly, the `kube-system` namespace of a cluster must not be altered or have any user-induced behaviors that could interfere with the auto-upgrade functionality.
```
